### DISCUSSION TITLE
Compare different web-os: node-os, OS13k, web3os, meeseOS etc... for os-js

### DISCUSSION THREAD

**Discussion by ghost** [os-js/OS.js#822] [Q&A]
Hi all.

#### 1. concept
I would like to compare different web-os. For example, [meeseOS/meeseOS](https://github.com/meeseOS/meeseOS), [github.com/web3os-org](https://github.com/web3os-org) and [os-js/OS.js](https://github.com/os-js/OS.js/) - I would like to know what features each of these projects has that is different and similar.

#### 2. Idea
I initially thought of creating a table comparing license type, development time and features. For example... 

#### Comparison of os.js(frontend, backend) to other similar solutions, alternatives

| os-name                                           |  license                          |  feature                                     | active?|written in web?      |
|---------------------------------------------------|-----------------------------------|----------------------------------------------|--------|---------------------|
| [OS.js](https://github.com/os-js/OS.js)           | BSD 2-Clause License              | full web-desktop                             | yes    | yes                 |
| [NodeOS](https://github.com/NodeOS/NodeOS)        | MIT License                       | desktop(only cli, without gui)               | no     | c, assembly,web     |
| [OS13k](https://github.com/KilledByAPixel/OS13k)  | GNU General Public License v3.0   | web desktop(but only games)                  | yes    | web                 |
| [meeseOS](https://github.com/meeseOS/meeseOS)     | MIT License                       | full web-desktop+games                       | yes    | web                 |
| [web3os](https://github.com/web3os-org)           | add                               | full web-desktop                             | yes    | asm, web            |
| [jde](https://github.com/Manjaro-WebDad/jde)      | GNU General Public License v3.0   | desktop envrionment for manjaro(distro-linux)| no     | web, c, python      |
| [Stealify](https://github.com/stealify/stealify)  | Apache License 2.0                | not is full desktop, compiler                | yes    | c, js               |
| [Awesome]()                                       | add                               | not is full desktop, compiler                | yes    | add.                |
| [Just-js](https://github.com/just-js)             | MIT License                       |  not is full desktop, compiler "partial"     | yes    | -                   |
| [B8G](https://github.com/stealify/b8g)            | The Unlicense                     | add.                                         | yes    | -                   |
| [Fuchsia OS](https://github.com/vsrinivas/fuchsia)| BSD 2-Clause "Simplified" License | full desktop.                                | yes    | c++,rust,go,python,c|
| [dahlia-os](https://github.com/dahliaOS/pangolin_desktop)|  Apache License 2.0        | full web-desktop.                            | yes    | dart, c, ruby, c... |

#### Other online operating systems / web desktops:

| Name & Link                                                                                           | Vague Description                | Repo & License                                                                                                                                         | Chat                                    |
|-------------------------------------------------------------------------------------------------------|----------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------|
| [![](images/start.png)&nbsp;Rahul.io](https://rahul.io/)                                              | another Windows 98 remake        | [repo here](https://github.com/lolstring/window98-html-css-js) - MIT license                                                                           |                                         |
| [![](images/icons/windows-93-16x16.png)&nbsp;Windows&nbsp;93](https://www.windows93.net/)             | a "fictional version of Windows" | (not open source currently)                                                                                                                            | [Discord](https://discord.gg/vAmMu7q)   |
| [🏡 ZineOS](https://whimsy.space/)                                                                     | emojis everywhere                | [repo here](https://github.com/STRd6/zine) - MIT license                                                                                               | [Discord](https://discord.gg/tUj4zYA)   |
| [![](images/icons/emuos-16x16.png)&nbsp;EmuOS](https://emupedia.net/beta/emuos/)                      | emulator-focused Win9X desktop   | [repo here](https://github.com/Emupedia/emupedia.github.io) - shared source                                                                            | [Discord](https://discord.gg/5u4VR9v)   |
| [![](images/icons/friend-16x16.png)&nbsp;Friend](https://friendup.cloud/)                             | another serious one              | [repo here](https://github.com/FriendUPCloud/friendup) - different licenses for different [parts](https://github.com/FriendUPCloud/friendup#licensing) |                                         |
| [![](images/icons/clouddesk-16x16.png)&nbsp;CloudDesk](http://altaica.altervista.org/)                | another serious one              | (explicitly not open source)                                                                                                                           |                                         |
| [![](images/icons/aaronos-16x16.png)&nbsp;AaronOS](https://aaronos.dev/AaronOS/aosBeta.php)           | interesting aesthetic            | [repo here](https://github.com/MineAndCraft12/AaronOS) - shared source                                                                                 | [Discord](https://discord.gg/eWDac7E)   |
| [![](images/icons/webdesktop-16x16.png)&nbsp;Web&nbsp;Desktop](https://webdesktop.net/)               | Windows 10 like "startpage"      |                                                                                                                                                        |                                         |
| [![](images/icons/system-16x16.png)&nbsp;System](https://system-developer-beta.000webhostapp.com/)    | Windows 10 like thing            |                                                                                                                                                        |                                         |
| [![](images/icons/winxp-doge-16x16.png)&nbsp;WinXP](https://winxp.now.sh/)                            | Windows XP                       | [repo here](https://github.com/ShizukuIchi/winXP/) - MIT license                                                                                       |                                         |
| [Windows 98 - Packard Belle](https://packard-belle.netlify.com/)                                      | Windows 98                       | [repo here](https://github.com/padraigfl/packard-belle-desktop) - shared source                                                                        |                                         |
| [X-WebDesktop-Vue](http://47.93.49.248/X-WebDesktop-Vue/)                                             | Windows 7                        | [repo here](https://github.com/OXOYO/X-WebDesktop-Vue) - MIT license                                                                                   |                                         |
| [![](images/icons/vue-win3.1-16x16.png)&nbsp;vue win3.1](https://disjfa.github.io/vue-win-3.1/)       | Windows 3.1                      | [repo here](https://github.com/disjfa/vue-win-3.1) - shared source                                                                                     |                                         |
| [![](images/icons/virtualdesktop.org-16x16.png)&nbsp;VirtualDesktop](http://virtualdesktop.org/)      | Screenshot flows, several OSes   |                                                                                                                                                        |                                         |
| [GUIdebook](https://guidebookgallery.org/guis/windows/)                                               | Screenshots, several OSes        |                                                                                                                                                        |                                         |

#### end-links
- source-code: [98.js.org](https://98.js.org/)
- source-code: [github/jaromaz/jslinux-mobile](https://github.com/jaromaz/jslinux-mobile) and demo [jm.iq.pl/jslinux-mobile/h7UNvdYq](https://jm.iq.pl/jslinux-mobile/h7UNvdYq/)
- demo: [rebornxp](https://rebornxp.js.org/), source-code: [RebornXP/rebornxp](https://github.com/RebornXP/rebornxp)
- source-code: [github/kurisubrooks/glass2](https://github.com/kurisubrooks/glass2) - Prototype HTML5 Desktop OS
- source-code: [github/JsOS-Team/JsOS](https://github.com/JsOS-Team/JsOS)
- source-code: [github/runtimejs/runtime](https://github.com/runtimejs/runtime)
- demo: [rajagopalan-gangadharan.github.io/JS-OS](https://rajagopalan-gangadharan.github.io/JS-OS/index.html), source-code: [github/RAJAGOPALAN-GANGADHARAN/JS-OS](https://github.com/RAJAGOPALAN-GANGADHARAN/JS-OS)
- source-code: [Maingron/MainOS](https://github.com/Maingron/MainOS)
- source-code: [UltimatePro-Grammer/websystem](https://github.com/UltimatePro-Grammer/websystem)
- demo: [mos](http://1j01.github.io/mos/)
- source-code: [Reactive-OS](https://github.com/mart-anthony-stark/Reactive-OS) and demo: [react-os](https://react-os.netlify.app/)
- source-code: [Artefact-Labs/KosmOS](https://github.com/Artefact-Labs/KosmOS)
- source-code: [screepers/ScreepsOS](https://github.com/screepers/ScreepsOS)
- source-code: [net-os.github.io/boot](https://net-os.github.io/boot/)
- source-code: [net-os/net-os.github.io](https://github.com/net-os/net-os.github.io)
- source-code: [juju2143/YukiOS](https://github.com/juju2143/YukiOS)

#### 3. question/feedback
*What do you all think about this idea?*

**Comment by ajmeese7** [comment]
I would like to note that "MeeseOS" (my repository) is simply a highly-customized fork of OS.js, so I wouldn't really consider it an entirely separate entity. I also have a highly personalized and simplified fork of AaronOS called "KaraOS" ([link](https://github.com/meese-enterprises/karameese.com)) that's a WIP (unfortunately abandoned for the time being due to time constraints) for an artistic portfolio site for my wife, that may be interesting to some folks (definitely wouldn't consider it a full OS). Just checked and it's up to 19k views now 😄 

I like the idea of the list, keep up the good work. So far it looks a lot like one of my star lists but with all the info compiled in a single place, which is neat and helpful to individuals window-shopping for a choice.

----

**Update:** Sorry if I gave the impression that I thought the list should be included in the OS.js repository, that was not what I was intending to say. See my [comment on the PR](https://github.com/os-js/OS.js/pull/823#issuecomment-1347364003) for my thoughts on that.

**Reply by ghost** [reply]
Hi ajmeese7, thank you for feedback.
